auth/basic/middleware.go

```diff
@@ -0,0 +1,92 @@
+package basic
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/subtle"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/go-kit/kit/endpoint"
+	httptransport "github.com/go-kit/kit/transport/http"
+)
+
+// AuthError represents generic Authorization error
```

Comment:
- [ ] s/represents generic Authorization/represents an authoriation/
- [ ] Punctuate this sentence.